core/src/main/java/org/mule/runtime/core/internal/routing/ScatterGatherRouter.java

```diff
@@ -36,7 +37,7 @@
  *
  * @since 3.5.0
  */
-public class ScatterGatherRouter extends AbstractForkJoinRouter {
+public class ScatterGatherRouter extends AbstractForkJoinRouter implements Router {
```

Comment:
Is it ok for this to be both a Scope and a Router? Does this have any impact on the extension model?

Reply:
I don't see why it would be an issue. Router is sub-type of Scope.